testing/integration/monitoring_endpoint_test.go

```diff
@@ -0,0 +1,154 @@
+
+//go:build integration
+
+package integration
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+
+	"github.com/elastic/elastic-agent-libs/kibana"
+	atesting "github.com/elastic/elastic-agent/pkg/testing"
+	"github.com/elastic/elastic-agent/pkg/testing/define"
+	"github.com/elastic/elastic-agent/pkg/testing/tools"
+	"github.com/elastic/elastic-agent/pkg/testing/tools/estools"
+	"github.com/elastic/elastic-agent/pkg/testing/tools/testcontext"
+)
+
+type EndpointMetricsMonRunner struct {
+	suite.Suite
+	info    *define.Info
+	fixture *atesting.Fixture
+}
+
+func TestEndpointAgentServiceMonitoring(t *testing.T) {
+	info := define.Require(t, define.Requirements{
+		Group: "fleet",
+		Stack: &define.Stack{},
+		Local: false, // requires Agent installation
+		Sudo:  true,  // requires Agent installation
+		OS: []define.OS{
+			{Type: define.Linux},
+		},
+	})
+
+	// Get path to agent executable.
+	fixture, err := define.NewFixtureFromLocalBuild(t, define.Version())
+	require.NoError(t, err, "could not create agent fixture")
+
+	runner := &EndpointMetricsMonRunner{
+		info:    info,
+		fixture: fixture,
+	}
+
+	suite.Run(t, runner)
+}
+
+func (runner *EndpointMetricsMonRunner) SetupSuite() {
+	deadline := time.Now().Add(10 * time.Minute)
+	ctx, cancel := testcontext.WithDeadline(runner.T(), context.Background(), deadline)
+	defer cancel()
+
+	runner.T().Log("Enrolling the agent in Fleet")
+	policyUUID := uuid.New().String()
+
+	createPolicyReq := kibana.AgentPolicy{
+		Name:        "test-policy-" + policyUUID,
+		Namespace:   "default",
+		Description: "Test policy " + policyUUID,
+		MonitoringEnabled: []kibana.MonitoringEnabledOption{
+			kibana.MonitoringEnabledLogs,
+			kibana.MonitoringEnabledMetrics,
+		},
+	}
+
+	installOpts := atesting.InstallOpts{
+		NonInteractive: true,
+		Force:          true,
+		Privileged:     true,
+	}
+
+	policy, err := tools.InstallAgentWithPolicy(ctx, runner.T(),
+		installOpts, runner.fixture, runner.info.KibanaClient, createPolicyReq)
+	require.NoError(runner.T(), err, "failed to install agent with policy")
+
+	runner.T().Log("Installing Elastic Defend")
+	pkgPolicyResp, err := installElasticDefendPackage(runner.T(), runner.info, policy.ID)
+	require.NoErrorf(runner.T(), err, "Policy Response was: %v", pkgPolicyResp)
+
+	runner.T().Log("Polling for endpoint-security to become Healthy")
+	ctx, cancel = context.WithTimeout(ctx, time.Minute*3)
+	defer cancel()
+
+	agentClient := runner.fixture.Client()
+	err = agentClient.Connect(ctx)
+	require.NoError(runner.T(), err, "could not connect to local agent")
+
+	require.Eventually(runner.T(),
+		func() bool { return agentAndEndpointAreHealthy(runner.T(), ctx, agentClient) },
+		time.Minute*3,
+		time.Second,
+		"Endpoint component or units are not healthy.",
+	)
+}
+
+func (runner *EndpointMetricsMonRunner) TestEndpointMetrics() {
```

Comment:
Is there a reason this can't just be another test in endpoint_security_test.go as well? 

Extending on that, could this just be sub-test added to the existing endpoint installation test to avoid having to install agent again just for this?

These are unit tests, they don't have to be completely isolated into their own test cases.

Reply:
So, I ended up making the test a bit more complex instead, to the point where it feels like it makes more sense to have it as its own test? You might disagree, though.